src/Widgets/PowerModeList.vala

```diff
@@ -0,0 +1,170 @@
+/*
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ * SPDX-FileCopyrightText: 2024 elementary, Inc. (https://elementary.io)
+ */
+
+using Power.Services.DBusInterfaces;
+
+public class Power.Widgets.PowerModeList : Gtk.Box {
+    public static bool successfully_initialized { get; private set; default = true; }
+
+    private static PowerProfile? pprofile;
+    private static GLib.Settings? settings;
+
+    public string settings_key { get; construct; }
+
+    private Gtk.RadioButton saver_radio;
+    private Gtk.RadioButton balanced_radio;
+    private Gtk.RadioButton performance_radio;
+
+    private const string ICON_RES = "/io/elementary/desktop/wingpanel/power/scalable/categories/";
+
+    public PowerModeList (bool on_battery) {
+        Object (
+            settings_key: on_battery ? "profile-on-good-battery" : "profile-plugged-in",
+            orientation: Gtk.Orientation.VERTICAL,
+            spacing: 0,
+            margin: 6
+        );
```

Comment:
The Object call here should really only contain things passed in as construction arguments. So it would only contain `on_battery`, but it seems `on_battery` is always false? so it's not clear why this is a construction argument at all

Reply:
It's a placehoder code, if it access the settings-daemon schema then it chooses the appropriate key to save/load the settings from. This is same as how it's there in the power switchboard plug.

Reply:
Implemented it, so no longer place holder.